### PR TITLE
Improve autocomplete for ui.input

### PR DESCRIPTION
**Pull request by phoskee**
Introducing case insensitive autocompletition!

### CODE REVIEW

File: nicegui/elements/input.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def toggle_type(_):
             def find_autocompletion() -> Optional[str]:
                 if self.value:
                     for item in autocomplete:
-                        if item.startswith(self.value):
+                        if item.lower().startswith(self.value.lower()):
```
Comment (rodja) [inline nicegui/elements/input.py:61]: Maybe you can `lower` the items only once in the constructor? That would save some cpu time...

Comment (rodja) [reply nicegui/elements/input.py:61]: Never mind. I thought we could do

```py
autocomplete = [item.lower() for item in autocomplete]
```
But then we lose the casing completely. It's better done in the way you coded it!

Comment (falkoschindler) [reply nicegui/elements/input.py:61]: But you can move `needle = self.value.lower()` out of the fow-loop.

**Comment by falkoschindler** [comment]
Thanks a lot for your countless experiments, @phoskee!
In order to get the case-insensitive autocompletion merged into main, I removed all other changes for now. And I switched to `casefold()` because this is basically made for "caseless matching".
Regarding autocompletion on mobile, I think we need to continue discussing solutions on https://github.com/zauberzeug/nicegui/discussions/852 or on a new PR demonstrating @rodja's datalist proposal.

**Pull request merged**